release/src/version-helpers.ts

```diff
@@ -86,14 +85,109 @@ export const getReleaseBranch = (versionString: string) => {
 };
 
 export const getVersionFromReleaseBranch = (branch: string) => {
+  const majorVersion = getMajorVersionNumberFromReleaseBranch(branch);
+
+  return `v0.${majorVersion}.0`;
+};
+
+export const getSdkVersionFromReleaseBranchName = async ({
+  github,
+  owner,
+  repo,
+  branchName,
+}: GithubProps & { branchName: string }) => {
+  const majorVersion = getMajorVersionNumberFromReleaseBranch(branchName);
+
+  let sdkVersion: string;
+
+  console.log(
+    `Resolved latest major release version - ${Number(majorVersion)}`,
+  );
+
+  console.log(
+    `Looking for git tag - "embedding-sdk-0.${Number(majorVersion)}.*"`,
+  );
+
+  const latestSdkTagForMajorRelease = await getLastEmbeddingSdkReleaseTag({
+    github,
+    owner,
+    repo,
+    majorVersion,
+  });
+
+  console.log(
+    `Resolved SDK latest release tag for v${majorVersion} - ${latestSdkTagForMajorRelease}`,
+  );
+
+  if (latestSdkTagForMajorRelease) {
+    const match = /embedding-sdk-(0\.\d+\.\d+(-nightly)?)/.exec(
+      latestSdkTagForMajorRelease,
+    );
+
+    if (match) {
+      sdkVersion = match[1];
+
+      console.log(
+        `Resolved SDK latest release version for v${majorVersion} - ${sdkVersion}`,
+      );
+
+      return sdkVersion;
+    }
+  }
+
+  const latestSdkTag = await getLastEmbeddingSdkReleaseTag({
+    github,
+    owner,
+    repo,
+  });
+
+  const match = /embedding-sdk-(0\.\d+\.\d+(-nightly)?)/.exec(latestSdkTag);
+  if (match) {
+    sdkVersion = match[1];
+
+    console.warn(
+      `Failed to resolve latest SDK package version! Using latest SDK version available - ${sdkVersion}`,
+    );
+
+    return sdkVersion;
+  }
```

Comment:
nit: might be nice to add some tests for the regex matches & version sorting function